### PR TITLE
Always drop the final hash in peer responses

### PR DESCRIPTION
**Pull request by teor2345**
To workaround a zcashd bug that squashes responses together.

### CODE REVIEW

**Comment by teor2345** [comment]
The clippy errors are unrelated to this PR, see #994.

File: zebrad/src/commands/start/sync.rs [hunk 1]
```diff
@@ -198,6 +198,24 @@ where
...
+                    // out-of-order first hashes.
+
+                    // We use the last hash for the tip, and we want to avoid bad
+                    // tips. So we discard the last hash. (We don't need to worry
+                    // about missed downloads, because we will pick them up again
+                    // in ExtendTips.)
+                    let hashes = match hashes.split_last() {
+                        None => continue,
+                        Some((_last, rest)) => rest,
+                    };
```
Comment (yaahc) [inline zebrad/src/commands/start/sync.rs]: similarly, slice pattern might look nicer here, style choice tho, I'll leave that up to you.

**Review by yaahc — DISMISSED** [review]

File: zebrad/src/commands/start/sync.rs (hunk 1 above)
Comment (teor2345) [reply zebrad/src/commands/start/sync.rs]: Refactored in 7ab0b0bf7745a5ced82529ec2d84f9af1d07c984

**Review by yaahc — APPROVED** [review]

File: zebrad/src/commands/start/sync.rs [hunk 2]
```diff
@@ -370,19 +401,21 @@
         Ok(())
     }
 
-    /// Queue downloads for each block that isn't currently known to our node
+    /// Queue download and verify tasks for each block that isn't currently known to our node
     async fn request_blocks(&mut self, hashes: Vec<block::Hash>) -> Result<(), Report> {
         tracing::debug!(hashes.len = hashes.len(), "requesting blocks");
         for hash in hashes.into_iter() {
-            // TODO: remove this check once the sync service is more reliable
```
Comment (hdevalence) [inline zebrad/src/commands/start/sync.rs:377]: I don't think the new comment addresses the concern of the old comment, which is that if we are queuing download of blocks we've already verified, something has gone wrong in the sync process.

The problem isn't that the check needs a better explanation, it's that the check failing indicates an invariant violation.

Comment (teor2345) [reply zebrad/src/commands/start/sync.rs:377]: I'll try to answer this question on #980

**Pull request merged**